Utilities/rollup/plugin-generate-refrences.js

```diff
@@ -0,0 +1,35 @@
+import * as fs from 'fs';
+import path from 'path';
+
+export function generateDtsReferences(pluginOptions) {
+
+  const rootDts = 'index.d.ts';
+  const dtsReferences = [
+    '/// <reference path="./types.d.ts" />',
+    '/// <reference path="./interfaces.d.ts" />',
+  ];
+
+  return {
+    name: 'generate-references',
+    generateBundle(outputOptions, bundle, isWrite) {
+      const files = Object.keys(bundle);
+
+      for (let i = 0; i < files.length; i++) {
+        const file = files[i];
+        const filename = file.replace('.js', '.d.ts');
+
+        const dts = path.join(pluginOptions['dest'], filename);
+
+        if (fs.existsSync(dts) && filename !== rootDts) {
+          dtsReferences.push(`/// <reference path="./${filename}" />`);
```

Comment:
I would add `filename.replace(/\\/g, '/')` here, since Windows will use backslashes for the path. This is minor, since we build and publish from a Unix system.